verilog/tools/extractor/verilog_extractor.cc

```diff
@@ -0,0 +1,332 @@
+                auto innerBlock = Extract(verible::SymbolCastToLeaf(*child), base);
+                blocks.insert(blocks.end(), innerBlock.begin(), innerBlock.end());
+            }
+        }
+    }
+
+    std::cout << "End Node" << std::endl << std::endl;
+    return blocks;
+}
+
+const verible::Symbol *GetChildByTag(const verible::SyntaxTreeNode &root, verilog::NodeEnum tag) {
+    for (const auto &child : root.children()) {
+        if (child) {
+            if (child->Tag().tag == static_cast<int>(tag)) {
+                return child.get();
+            }
+        }
+    }
+    return nullptr;
+}
+
+const verible::Symbol *GetFirstChildByTag(const verible::SyntaxTreeNode &root, verilog::NodeEnum tag) {
+    auto target = GetChildByTag(root, tag);
+    if (target == nullptr) {
+        for (const auto &child : root.children()) {
+            if (child && child->Kind() == verible::SymbolKind::kNode) {
+                target = GetFirstChildByTag(verible::SymbolCastToNode(*child), tag);
+                if (target != nullptr) {
+                    return target;
+                }
+            }
+        }
+    }
+    return target;
+}
+
+Block ExtractModuleInstantiation(const verible::SyntaxTreeNode &node, const verible::SyntaxTreeNode &root,
+                                 absl::string_view base) {
+
+    auto instantiationBase = GetChildByTag(node, verilog::NodeEnum::kInstantiationBase);
+
+    auto instantiationType = GetChildByTag(verible::SymbolCastToNode(*instantiationBase),
+                                           verilog::NodeEnum::kInstantiationType);
+
+    auto type = verible::SymbolCastToLeaf(
+            *GetFirstChildByTag(verible::SymbolCastToNode(*instantiationType),
+                                verilog::NodeEnum::kNetVariableDeclarationAssign));
+
+    Anchor typeAnchor = Anchor(type, base);
+
+    auto variableList = GetChildByTag(verible::SymbolCastToNode(*instantiationBase),
+                                      verilog::NodeEnum::kGateInstanceRegisterVariableList);
+    auto variableName = verible::SymbolCastToLeaf(
+            *GetFirstChildByTag(verible::SymbolCastToNode(*variableList),
+                                verilog::NodeEnum::kNetVariableDeclarationAssign));
+    Anchor variableNameAnchor = Anchor(variableName, base);
+
+    Block moduleInstance = Block(Type::ModuleInstance);
+    moduleInstance.AppendAnchor(typeAnchor);
+    moduleInstance.AppendAnchor(variableNameAnchor);
+
+    return moduleInstance;
+}
+
+Anchor
+ExtractModuleEnd(const verible::SyntaxTreeNode &node, const verible::SyntaxTreeNode &root, absl::string_view base) {
+    auto moduleEndKeyword = verible::SymbolCastToLeaf(
+            *GetChildByTag(node, verilog::NodeEnum::kNetVariableDeclarationAssign));
+    Anchor moduleNameAnchor = Anchor(moduleEndKeyword, base);
+    return moduleNameAnchor;
+};
+
+Anchor
+ExtractModuleHeader(const verible::SyntaxTreeNode &node, const verible::SyntaxTreeNode &root, absl::string_view base) {
```

Comment:
The convention in verible code is to put & and * after the type, not as a prefix to the variable. Instead
`const verible::SyntaxTreeNode &node`
do
`const verible::SyntaxTreeNode& node`

The goal is to be consistent with the rest of the code base. The consistency makes the code more readable.
An example: in case the existing code doesn't fully comply with the agreed code style, the new code should follow that non-compliance (introduce the same deviation from the code style). The consistency trumps the code style.